### PR TITLE
Implemented advanced feed settings to allow cookies and HTTP redirects.

### PR DESCRIPTION
**Pull request by rdmitr**
Closes #

I have noticed that one of my feeds tries to perform an HTTP redirect while setting a cookie (some kind of spam/rate limiting thing I guess). Neither option is currently supported by FreshRSS, so I added them as advanced feed options.

Changes proposed in this pull request:

- Add the advanced feed options to allow cookies and HTTP redirects. Both are disabled by default.

How to test the feature manually:

1. When subscribing to a feed, open the "Advanced" settings menu and set the "Allow cookies" option, and a value of 5 for HTTP redirects;
2. Subscribe to the feed;
3. In the "feed" database table, check the "attributes" column. Observe the following attributes:
    1. "10031": "" (`CURLOPT_COOKIEFILE`)
    2. "68": 5 (`CURLOPT_MAXREDRS`)
    3. "52": 1 (`CURLOPT_FOLLOWLOCATION`)

A similar sequence of actions can be performed to manually test modification of these attributes on an existing feed.

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [ ] unit tests written (optional if too hard)
- [ ] documentation updated

### CODE REVIEW

File: app/i18n/fr/sub.php [hunk 1]
```diff
@@ -36,6 +36,8 @@
 		'title' => 'Titre',
 	),
 	'feed' => array(
+		'accept_cookies' => 'Accept cookies',	// TODO
+		'accept_cookies_help' => 'Allow the feed server to set cookies (stored in memory for the duration of the request only)',	// TODO
```
Comment (Alkarex) [inline app/i18n/fr/sub.php]:
```suggestion
		'accept_cookies' => 'Autoriser les cookies',
		'accept_cookies_help' => 'Accepte les cookies du flux (stocké en mémoire seulement le temps de la requête)',
```

File: app/i18n/fr/sub.php [hunk 2]
```diff
@@ -116,6 +118,8 @@
 			'reload_articles_help' => 'Recharge cette quantité d’articles et récupère le contenu complet si un sélecteur est défini.',
 			'title' => 'Maintenance',	// IGNORE
 		),
+		'max_http_redir' => 'Max HTTP redirects',	// TODO
+		'max_http_redir_help' => 'Set to 0 or leave blank to disable, -1 for unlimited redirects',	// TODO
```
Comment (Alkarex) [inline app/i18n/fr/sub.php]:
```suggestion
		'max_http_redir' => 'Maximum de redirections HTTP',
		'max_http_redir_help' => 'Mettre à 0 ou vide pour désactiver, -1 pour un nombre illimité de redirections',
```

File: app/i18n/fr/sub.php (hunk 1 above)
Comment (Alkarex) [inline app/i18n/fr/sub.php]:
```suggestion
		'accept_cookies' => 'Autoriser les cookies',
		'accept_cookies_help' => 'Accepte les cookies du flux (stocké en mémoire seulement le temps de la requête)',
```

File: app/views/subscription/add.phtml [hunk 3]
```diff
@@ -172,6 +172,27 @@
 				</div>
 			</div>
 
+			<div class="form-group">
+				<label class="group-name" for="curl_params_cookie"><?= _t('sub.feed.css_cookie') ?></label>
+				<div class="group-controls">
+					<input type="text" name="curl_params_cookie" id="curl_params_cookie" class="extend" value="" placeholder="<?= _t('gen.short.blank_to_disable') ?>" />
```
Comment (Alkarex) [inline app/views/subscription/add.phtml:178]: Why do you use `curl_params_cookie` in subscription/add but `curl_params_cookiefile` in feed/update?
Shouldn't you have a single parameter to enable cookies?

Comment (rdmitr) [reply app/views/subscription/add.phtml:178]: In feed/update, there already was the `curl_params_cookie` parameter which allows setting custom user-specified cookies. I added the `curl_params_cookiefile` parameter to pass the `CURLOPT_COOKIEFILE` option, which enables the libcurl cookie engine.

On the other hand, in subscription/add, the `curl_params_cookie` setting was not available, so I copied it over from feed/update for consistency, and added `curl_params_cookiefile` as well. The meaning is the same: `curl_params_cookie` for user-specified cookies, `curl_params_cookiefile` to enable the libcurl cookie engine.

**Comment by Alkarex** [comment]
Looks all good! Sorry for the delay, and thanks again 👍🏻 
Please add a line for you in https://github.com/FreshRSS/FreshRSS/blob/edge/CREDITS.md

**Pull request merged**